### PR TITLE
[feature] update related flows and implement backfill one-time job for users' `totalDonationAmountUsd`

### PR DESCRIPTION
**Pull request by benson-budiman-cb**
relates to #660 

## What changed? Why?

This PR updates the related flows that involve some form of donation (_e.g._ Coinbase Commerce donations and user-funded mints) and implements one-time backfill job for the users' `totalDonationAmountUsd` column. Why? Because we now want to populate this field appropriately, but we also want to periodically make sure that each rows' is correct.

## UI changes

No UI changes.

## PlanetScale Deploy Request

No PlanetScale schema changes.

## Notes to reviewers

This PR changes the following:
- The PR increments the `totalDonationAmountUsd` column in the Coinbase Commerce webhook receiver, in the NFT mint server action, and in the "merge users" logic
- The seeding function also includes addition of the `totalDonationAmountUsd` column
- The PR also implements a one-time script that backfills all the users' `totalDonationAmountUsd` in batches
  - See [this comment](https://github.com/Stand-With-Crypto/swc-web/pull/718/files#diff-dbe265101a94a45373e7463926e16f6aaa9a543640fce9f4daa8a1f5672a282dR22-R36) for the Inngest function implementation choices

## How has it been tested?

- [X] Locally
- [X] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

https://github.com/Stand-With-Crypto/swc-web/assets/135282747/c10a639a-c538-4fe6-922d-3924a9ba6c6c

Screenshot 2024-03-27 at 10 52 40 PM

Screenshot 2024-03-27 at 10 52 52 PM

Screenshot 2024-03-27 at 10 53 05 PM

## Change management

type=routine 
risk=low 
impact=sev5

### CODE REVIEW

File: src/bin/seed/seedLocalDb.ts [hunk 1]
```diff
@@ -326,6 +326,31 @@ async function seed() {
   const userAction = await prismaClient.userAction.findMany()
   logEntity({ userAction })
 
+  const userActionsWithNftMint = await prismaClient.userAction.findMany({
```
Comment (travisbloom-cb) [inline src/bin/seed/seedLocalDb.ts:329]: we fetch a few lines above, just add your include and filter in-memory?

File: src/bin/seed/seedLocalDb.ts [hunk 2]
```diff
@@ -411,6 +436,30 @@ async function seed() {
   const userActionDonation = await prismaClient.userActionDonation.findMany()
   logEntity({ userActionDonation })
 
+  const userActionsWithDonation = await prismaClient.userAction.findMany({
```
Comment (travisbloom-cb) [inline src/bin/seed/seedLocalDb.ts:439]: we fetch a few lines above, just add your include and filter in-memory?

File: src/inngest/functions/auditUsersTotalDonationAmountUsdCronJob/index.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+const AUDIT_USERS_TOTAL_DONATION_AMOUNT_USD_MINI_BATCH_SIZE =
+  Number(process.env.AUDIT_USERS_TOTAL_DONATION_AMOUNT_USD_MINI_BATCH_SIZE) || 100
+
+const UPDATE_USER_BATCH_FUNCTION_ID =
+  'script.audit-users-total-donation-amount-usd.update-batch-of-users'
+const UPDATE_USER_BATCH_EVENT_NAME =
+  'script/audit.users.total.donation.amount.usd/update.batch.of.users'
+
+export const auditUsersTotalDonationAmountUsdInngestCronJobUpdateBatchOfUsers =
```
Comment (travisbloom-cb) [inline src/inngest/functions/auditUsersTotalDonationAmountUsdCronJob/index.ts:92]: we should never need to fix users, so if any user does need to be updated, it should trigger a sentry error (maybe we add after the initial backfill is run)

**Review by travisbloom-cb — CHANGES_REQUESTED** [review]

File: src/inngest/functions/auditUsersTotalDonationAmountUsdCronJob/index.ts (hunk 3 above)
Comment (benson-budiman-cb) [reply src/inngest/functions/auditUsersTotalDonationAmountUsdCronJob/index.ts:92]: @travisbloom-cb ah okay, I think I misunderstood the purpose of the auditing cron job then. I will create a separate Inngest function that serves as the one-time backfill, and then repurpose this current job as the one to trigger Sentry messages.

File: src/bin/seed/seedLocalDb.ts (hunk 1 above)
Comment (benson-budiman-cb) [reply src/bin/seed/seedLocalDb.ts:329]: good idea

File: src/bin/seed/seedLocalDb.ts (hunk 2 above)
Comment (benson-budiman-cb) [reply src/bin/seed/seedLocalDb.ts:439]: good idea

File: src/inngest/functions/auditUsersTotalDonationAmountUsdCronJob/index.ts (hunk 3 above)
Comment (benson-budiman-cb) [reply src/inngest/functions/auditUsersTotalDonationAmountUsdCronJob/index.ts:92]: I'll put the cron job in a separate PR.

**Review by travisbloom-cb — APPROVED** [review]

**Pull request merged**